src/ripple/app/tests/Env_test.cpp

```diff
@@ -0,0 +1,199 @@
+//------------------------------------------------------------------------------
+/*
+  This file is part of rippled: https://github.com/ripple/rippled
+  Copyright (c) 2012-2015 Ripple Labs Inc.
+
+  Permission to use, copy, modify, and/or distribute this software for any
+  purpose  with  or without fee is hereby granted, provided that the above
+  copyright notice and this permission notice appear in all copies.
+
+  THE  SOFTWARE IS PROVIDED "AS IS" AND THE AUTHOR DISCLAIMS ALL WARRANTIES
+  WITH  REGARD  TO  THIS  SOFTWARE  INCLUDING  ALL  IMPLIED  WARRANTIES  OF
+  MERCHANTABILITY  AND  FITNESS. IN NO EVENT SHALL THE AUTHOR BE LIABLE FOR
+  ANY  SPECIAL ,  DIRECT, INDIRECT, OR CONSEQUENTIAL DAMAGES OR ANY DAMAGES
+  WHATSOEVER  RESULTING  FROM  LOSS  OF USE, DATA OR PROFITS, WHETHER IN AN
+  ACTION  OF  CONTRACT, NEGLIGENCE OR OTHER TORTIOUS ACTION, ARISING OUT OF
+  OR IN CONNECTION WITH THE USE OR PERFORMANCE OF THIS SOFTWARE.
+*/
+//==============================================================================
+
+#include <BeastConfig.h>
+#include <ripple/app/tests/Env.h>
+#include <ripple/json/to_string.h>
+#include <ripple/protocol/TxFlags.h>
+#include <beast/unit_test/suite.h>
+
+namespace ripple {
+namespace test {
+
+class Env_test : public beast::unit_test::suite
+{
+public:
+    // Signing with secp256k1 and ed25519 keys
+    void
+    testKeyType()
+    {
+        using namespace jtx;
+
+        Env env(*this);
+        Account const alice("alice", KeyType::ed25519);
+        Account const bob("bob", KeyType::secp256k1);
+        Account const carol("carol");
+        env.fund(XRP(10000), alice, bob);
+
+        // Master key only
+        env(noop(alice));
+        env(noop(bob));
+        env(noop(alice), sig("alice"),                          ter(tefBAD_AUTH_MASTER));
+        env(noop(alice), sig(Account("alice",
+            KeyType::secp256k1)),                               ter(tefBAD_AUTH_MASTER));
+        env(noop(bob), sig(Account("bob",
+            KeyType::ed25519)),                                 ter(tefBAD_AUTH_MASTER));
+        env(noop(alice), sig(carol),                            ter(tefBAD_AUTH_MASTER));
+
+        // Master and Regular key
+        env(regkey(alice, bob));
+        env(noop(alice));
+        env(noop(alice), sig(bob));
+        env(noop(alice), sig(alice));
+
+        // Regular key only
+        env(set(alice, asfDisableMaster), sig(alice));
+        env(noop(alice));
+        env(noop(alice), sig(bob));
+        env(noop(alice), sig(alice),                            ter(tefMASTER_DISABLED));
+        env(clear(alice, asfDisableMaster), sig(alice),         ter(tefMASTER_DISABLED));
+        env(clear(alice, asfDisableMaster), sig(bob));
```

Comment:
Add one more test here to verify that the `asfDisableMaster` flag was actually cleared:
```
env(noop(alice), sig(alice));
```